docs/specs/app-initiated-sso-to-web.md

```diff
@@ -0,0 +1,179 @@
+
+Step [8] redirects the browser to Web App #2, with a session cookies set. The session cookie can then be used for authorizing Web App #2 as it was logged in as the same user of Mobile App #1.
+
+## Native App Authorization Extensions
+
+Same as the [Native SSO](https://openid.net/specs/openid-connect-native-sso-1_0.html#name-native-app-authorization-ex), except that the new scope value supported is `x_device_browser_sso`.
+
+## Token Exchange Profile for App Initiated SSO to Web
+
+This section profiles the OAuth2 Token Exchange [RFC8693] spec and describes the processing rules used to exchange a previous authentication for a device browser session token, which can then be used to exchange for a browser session of another Web App.
+
+### Defined token type identifiers
+
+This profile also defines the following token type identifiers.
+
+```
+urn:x-oath:params:oauth:token-type:device-secret
+```
+
+Please refer to [OpenID Connect Native SSO for Mobile Apps 1.0](https://openid.net/specs/openid-connect-native-sso-1_0.html).
+
+```
+urn:authgear:params:oauth:token-type:device-browser-session-token
+```
+
+This token type identifier refers to the device browser session token mentioned in this spec. It can be used as the `requested_token_type` parameter in token exchange request.
+
+### Token Exchange Request
+
+When a mobile app wants to initiate SSO to web, it makes a /token endpoint request to obtain a device browser session token with the following parameters:
+
+- `grant_type`: Must be `urn:ietf:params:oauth:grant-type:token-exchange`
+- `subject_token`: The id_token obtained by the mobile app
+- `subject_token_type`: Must be `urn:ietf:params:oauth:token-type:id_token`
+- `actor_token`: The device_secret obtained by the mobile app
+- `actor_token_type`: Must be `urn:x-oath:params:oauth:token-type:device-secret`
+- `requested_token_type`: Optional. If given, the value must be `urn:authgear:params:oauth:token-type:device-browser-session-token`.
+
+Here is an example:
+
+```
+POST /token HTTP/1.1
+Host: auth.example.com
+...
+grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange
+&subject_token=<id_token>
+&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aid-token
+&actor_token=95twdf3w4y6wvftw35634t
+&actor_token_type=urn%3Ax-oath%3Aparams%3Aoauth%3Atoken-type%3Adevice-secret
+&requested_token_type=urn%3Aauthgear%3Aparams%3Aoauth%3Atoken-type%3Adevice-browser-session-token
+```
+
+### Token Exchange Response
+
+The Token Exchange response for this profile has the following characteristics:
+
+- `device_secret`: Required. A new device secret must be returned. The mobile app should update its device secret with the new secret.
+- `x_device_browser_session_token`: Required. The device browser session token which can then be used to exchange for a browser cookie.
+- `expires_in`: Required. When the device browser session token expires.
```

Comment:
The token response is not compliant to the spec. https://datatracker.ietf.org/doc/html/rfc8693#name-successful-response We must include `access_token`, `issued_token_type `, and `token_type`.

> access_token
REQUIRED. The security token issued by the authorization server in response to the token exchange request. The access_token parameter from [Section 5.1](https://www.rfc-editor.org/rfc/rfc6749#section-5.1) of [[RFC6749](https://datatracker.ietf.org/doc/html/rfc6749)] is used here to carry the requested token, which allows this token exchange protocol to use the existing OAuth 2.0 request and response constructs defined for the token endpoint. The identifier access_token is used for historical reasons and the issued token need not be an OAuth access token.

Although the key is `access_token`, it does not have to be an OAuth access token. In our case the token response should be

```
{
  "access_token": "DEVICE_BROWSER_SESSION_TOKEN",
  "issued_token_type": "urn:authgear:params:oauth:token-type:device-browser-session-token",
  "token_type": "DPoP"
}
```

Note that the `token_type` is `DPoP`, if DPoP is used. Otherwise, it is `Bearer`. This also applies to any `token_type` we return in other kind of Token responses.